app/src/main/java/com/amaze/filemanager/adapters/RecyclerAdapter.java

```diff
@@ -176,6 +177,11 @@ public RecyclerAdapter(
    * @param imageView the check {@link CircleGradientDrawable} that is to be animated
    */
   public void toggleChecked(int position, ImageView imageView) {
+    if (itemsDigested.size() <= position) {
```

Comment:
Why is the position larger than the item size? are there too few items or does something calculate the item clicked incorrectly?

Reply:
Item position is directly being fetched from `viewholder.getAdapterPosition` in `onBindViewHolder`
Not sure why it's returning position greater than the recyclerview adapter size.

Reply:
But @VishalNehra, I have a concern, as @EmmanuelMess suggests, the error shows `java.lang.ArrayIndexOutOfBoundsException: length=22; index=-1` means the requested index is `-1` and not something greater than the length of the array!

Reply:
Very well, can put a check for index < 0 as well.